volatility3/plugins/windows/notepad.py

```diff
@@ -0,0 +1,707 @@
+                     0x70d9, 0x72e0, 0x72f9, 0x7360, 0x7368, 0x74f7, 0x766e, 0x7690, 0x76b1, 0x76ba, 0x7852, 0x7940,
+                     0x7d0a, 0x7f87, 0x7fa8, 0x814a, 0x852c, 0x863c, 0x87ba, 0x88f4, 0x8ba5, 0x8bb6, 0x8e81, 0x94b5,
+                     0x9890, 0x9a6d, 0x9a7e, 0x9a8f, 0x9aa7, 0x9ab8, 0x9ad2, 0x9ae3, 0x9af4, 0x9c48, 0x9ed6, 0x9f2c,
+                     0xaf2c, 0xaf34, 0xb080, 0xb220, 0xb550, 0xb69c, 0xbc1f, 0xd329}
+
+
+class Notepad(interfaces.plugins.PluginInterface):
+    """Dumps displayed text from open Notepad processes"""
+
+    _required_framework_version = (2, 0, 0)
+    _version = (1, 0, 0)
+
+    @classmethod
+    def get_requirements(cls) -> List[interfaces.configuration.RequirementInterface]:
+        return [
+            requirements.ModuleRequirement(
+                name="kernel",
+                description="Windows kernel",
+                architectures=["Intel32", "Intel64"],
+            ),
+            requirements.VersionRequirement(name="pslist", component=pslist.PsList, version=(2, 0, 0)),
+            requirements.VersionRequirement(name="vadinfo", component=vadinfo.VadInfo, version=(2, 0, 0)),
+            requirements.ListRequirement(
+                name="pid",
+                element_type=int,
+                description="Manually specifies notepad process IDs",
+                optional=True,
+            ),
+        ]
+
+    def _generator(self, procs):
+        kernel = self.context.modules[self.config["kernel"]]
+
+        protect_values = vadinfo.VadInfo.protect_values(
+            self.context,
+            kernel.layer_name,
+            kernel.symbol_table_name
+        )
+
+        for proc in procs:
+            proc_id = "Unknown"
+            try:
+                proc_id = proc.UniqueProcessId
+                proc_layer_name = proc.add_process_layer()
+            except exceptions.InvalidAddressException as ex:
+                vollog.debug(
+                    "Process {}: invalid address {} in layer {}".format(
+                        proc_id, ex.invalid_address, ex.layer_name
+                    )
+                )
+                continue
+
+            proc_layer = self.context.layers[proc_layer_name]
+
+            for vad in sorted(list(vadinfo.VadInfo.list_vads(proc)), key=lambda x: x.get_start()):
+                vad_start = vad.get_start()
+                vad_size = vad.get_size()
+                protection = vad.get_protection(
+                    protect_values,
+                    vadinfo.winnt_protections
+                )
+                private = int(vad.get_private_memory())
+                tag = vad.get_tag()
+
+                if private and protection == 'PAGE_READWRITE' and vad_size < 16777216 and tag in ('VadS', 'Vadm'):
+                    try:
+                        data = proc_layer.read(vad_start, vad_size, pad=True)
+                        if not data:
```

Comment:
You're padding, so this will always return a binary string of `vad_size`, so this check should'nt ever fire unless `vad_size` is 0, which you could check before attempting the read, so this doesn't seem useful.

Reply:
Oh, I see, that makes sense. It was part of the code from the VadInfo plugin that read the VADs in chunks. I've removed the chunking and forgot to this check, so it wasn't intentional, and even if it would fire, it would break out of the loop over the VAD list, which I certaintly didn't intend to do.